packages/flutter/lib/src/widgets/router.dart

```diff
@@ -471,22 +468,26 @@ class Router<T> extends StatefulWidget {
 typedef _AsyncPassthrough<Q> = Future<Q> Function(Q);
 typedef _DelegateRouteSetter<T> = Future<void> Function(T);
 
-// Whether to report the route information in this build cycle.
-enum _IntentionToReportRouteInformation {
-  // We haven't receive any signal on whether to report.
-  none,
-  // Report if route information changes.
-  maybe,
-  // Report regardless of route information changes.
-  must,
-  // Don't report regardless of route information changes.
-  ignore,
+/// The [Router]'s intention when it reports a new [RouteInformation] to the
+/// [RouteInformationProvider].
+///
+/// See also:
+///
+///  * [RouteInformationProvider.routerReportsNewRouteInformation]: which is
+///    called by the router when it has a new route information to report.
+enum RouteInformationReportingType {
```

Comment:
nit: the names in this enum are inconsistent: implicit is an adjective, neglecting as a verb form, and navigation is a noun. We should probably be more consistent here.

Reply:
Maybe just neglect and navigate for the last two. Not exactly sure what we want to indicate with "implicit", so not sure what to use there.